### PR TITLE
[#157] Webcam stream

### PR DESCRIPTION
**Pull request by pobudzey**
# General info

**Issue Number**: #157

**Task description**: 

 - Made some adjustments to device, encoder and decoder creation
 - Added an option to stream from webcam instead of selecting a file

# Testing

**Test coverage:** n/a

# Additional notes

**Note to reviewers**:
- The webcam used by the sample sender is defined in a configuration file (config.json)
- To figure out the name of your webcam(s), run the following command
`ffmpeg -list_devices true -f dshow -i dummy`
- If both a file is selected AND the "Use Camera" option is checked, the sender will use the camera by default

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,8 +15,11 @@ Sample senders and receivers to be used with the Switchboard project. https://gi
 2. Change directory to src  
 `cd src`
 
-3. Run the video receiver  
+3. Edit the configuration file if necessary  
+`vi config.json`
```
Comment (felixlapierre) [inline README.md]: We should include an extra blurb about how
- you can edit the config file to choose which camera the sender will use
- the ffmpeg command to use to get the name of the cameras you have
basically add the explanation in this PR to the readme.

File: src/receiver.py [hunk 2]
```diff
@@ -68,8 +68,10 @@ def consume_stream(self, id):
                 stream_info["inputChannel"]["decoder"]["serialNumber"]
                 == self.serial_number
             ):
-                ip = stream_info["outputChannel"]["encoder"]["device"]["ipAddress"]
-                port = stream_info["inputChannel"]["port"]
+                ip = stream_info["outputChannel"]["encoder"]["device"][
+                    "privateIpAddress"
+                ]
+                port = stream_info["inputChannel"]["channel"]["port"]
```
Comment (felixlapierre) [inline src/receiver.py:74]: I had to add the ["channel"] here for it to work on my end.

**Review by felixlapierre — COMMENTED** [review]
I made a small fix to match the data sent by the backend on my branch. Let me know if that's fine. Otherwise it works great on my end LGTM

**Comment by felixlapierre** [comment]
i think we can ignore the duplication check

File: README.md (hunk 1 above)
Comment (pobudzey) [reply README.md]: Agreed, done

**Review by felixlapierre — APPROVED** [review]

**Pull request merged**